apps/dashboard/app/javascript/path_selector/path_selector_data_table.js

```diff
@@ -225,4 +227,17 @@ export class PathSelectorTable {
     data.files = filteredFiles;
     return data;
   }
+
+  filteredByFilename(file) {
+    if (this.filePattern !== "") {
+      if (file.name.match(RegExp(this.filePattern))) {
```

Comment:
I think this can throw an exception if `this.filePattern` doesn't compile correctly. Seems like we should catch that exception.